macros/scaffold.pl

```diff
@@ -495,7 +508,28 @@ sub Begin {
   Scaffold->Error("Sections must appear within a Scaffold") unless $scaffold;
   Scaffold->Error("Section::Begin() while a section is already open") if $scaffold->{current_section};
   my $self = $scaffold->start_section(Section->new(@_));
-  $self->{name} = "Part $self->{number}:" unless $self->{name};
+  my $number_at_depth;
+  if ($scaffold->{depth} == 1) {
+	$number_at_depth = ('a'..'zz')[$self->{number}-1];
+  } elsif ($scaffold->{depth} == 2) {
+	# Avoiding a package for roman numerals
+	my @romanatom = (['i','v','x'],['x','l','c'],['c','d','m']);
+	my @romanmolecule = map{['',$_->[0],$_->[0]x2,$_->[0]x3,$_->[0].$_->[1],$_->[1],$_->[1].$_->[0],$_->[1].$_->[0]x2,$_->[1].$_->[0]x3,$_->[0].$_->[2]]}(@romanatom);
+	my @roman;
+	for my $i (@{$romanmolecule[2]}) {
+		for my $j (@{$romanmolecule[1]}) {
+			for my $k (@{$romanmolecule[0]}) {
+				push(@roman,$i.$j.$k);
+			}
+		}
+	}
```

Comment:
Do you really anticipate having thousands of scaffolds nested at depth 2?  This code seems excessive.

Reply:
Since I went `a..zz` at level 1, this was for consistency to use about the same order of magnitude. 26^2 compared to 999. No, no one will ever need that many.

Reply:
Now I do wonder if it's worth producing arrays that are 26^2, 999,and 26^2 elements long, just to pick one entry from them.  I think I will add a conditional to check if the number is "small" first (which it will almost always be) and do something direct in that case.